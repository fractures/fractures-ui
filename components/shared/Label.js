import React, { Fragment } from "react"

const Label = props => (
	<Fragment>
		<span className="fr-label nowrap ellipsis">{props.children}</span>
		<style jsx global>{`
			.fr-label {
				color: var(--shade-5);

				font-size: 0.875rem;
				font-weight: 500;
				line-height: 1.75rem;
			}
		`}</style>
	</Fragment>
)

export default Label
