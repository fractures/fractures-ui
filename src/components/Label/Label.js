import classNames from "@sindresorhus/class-names"
import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	color: PropTypes.oneOf([null, "primary", "green", "red", "yellow", "blue"]),
	label: PropTypes.string.isRequired
}

const defaultProps = {
	color: null,
	label: ""
}

const NakedLabel = props => {
	const labelClass = classNames(props.className, {
		"fr-label--default": !props.color,
		[`fr-label--${ props.color }`]: props.color
	})

	return (
		<div className={ labelClass }>
			<span>{props.label}</span>
		</div>
	)
}

const Label = styled(NakedLabel)`
	position: relative;

	display: inline-flex;
	height: 1.5rem;
	padding: 0 0.5rem;

	border-radius: 0.125rem;
	clip-path: polygon(100% 0, 100% calc(100% - 6px), calc(100% - 6px) 100%, 0 100%, 0 0);

	font-size: 0.75rem;
	font-weight: 500;
	line-height: 1.5rem;

	&:after {
		position: absolute;
		right: 0;
		top: 1.125rem;

		height: 0.375rem;
		width: 0.375rem;

		background-color: rgba(0, 0, 0, 0.2);
		background-blend-mode: multiply;
		border-top-left-radius: 0.125rem;
		content: " ";
	}

	&.hl-label--default {
		background-color: var(--fr-100);
		color: var(--fr-700);
	}

	&.hl-label--primary {
		background-color: var(--fr-300);
		color: var(--fr-900);
	}

	&.hl-label--green span,
	&.hl-label--red span,
	&.hl-label--yellow span,
	&.hl-label--blue span {
		mix-blend-mode: multiply;
	}

	&.hl-label--green {
		background-color: var(--fr-color-green);
		color: rgba(0, 0, 0, 0.9);
	}

	&.hl-label--red {
		background-color: var(--fr-color-red);
		color: rgba(0, 0, 0, 0.9);
	}

	&.hl-label--yellow {
		background-color: var(--fr-color-yellow);
		color: rgba(0, 0, 0, 0.9);
	}

	&.hl-label--blue {
		background-color: var(--fr-color-blue);
		color: rgba(0, 0, 0, 0.9);
	}
`

NakedLabel.propTypes = propTypes
NakedLabel.defaultProps = defaultProps

Label.propTypes = propTypes
Label.defaultProps = defaultProps

export default Label
