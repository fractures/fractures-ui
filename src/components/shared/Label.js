import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired
}

const NakedLabel = ({ children }) => <span className="fr-label nowrap ellipsis">{children}</span>

const Label = styled(NakedLabel)`
	color: var(--fr-900);

	font-size: 0.875rem;
	font-weight: 500;
	line-height: 1.75rem;
`

Label.propTypes = propTypes

export default Label
