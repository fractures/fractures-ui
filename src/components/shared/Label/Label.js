import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
	className: PropTypes.string
}

const NakedLabel = ({ children, className }) => <span className={ className }>{children}</span>

const Label = styled(NakedLabel)`
	align-items: center;
	display: flex;
	white-space: nowrap;

	color: var(--color-gray-900);

	font-size: 0.875rem;
	font-weight: 500;
	line-height: 1.5rem;
	text-overflow: ellipsis;
`

NakedLabel.propTypes = propTypes

Label.propTypes = propTypes

export default Label
