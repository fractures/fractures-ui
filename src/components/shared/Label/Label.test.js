import Label from "./Label.js"

describe("Label", () => {
	const tree = renderer.create(<Label>Label</Label>).toJSON()

	it("Renders correctly", () => expect(tree).toMatchSnapshot())
})
