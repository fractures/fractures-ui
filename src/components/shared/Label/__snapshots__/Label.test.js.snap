// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Renders correctly 1`] = `
<span
  className="sc-bdVaJa hOTKmH"
>
  Label
</span>
`;
